### ISSUE TITLE
Rule: for-direction not working with '+=' syntax

### ISSUE THREAD

**Issue by da3monHunter** [eslint/eslint#12622]
* **ESLint Version: 6.7.1**
* **Node Version: 13.1.0**
* **npm Version: 6.12.1**
* **parser: Babel-ESLin**

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
module.exports = {
  "extends": [
    "airbnb",
    "prettier",
    "prettier/react",
    "plugin:prettier/recommended",
    "eslint-config-prettier"
  ],
  "parser": "babel-eslint",
  "rules": {
    "import/no-unresolved": "off",
    "no-console" : "warn",
    "for-direction": "error",
    "react/jsx-filename-extension": [
      1,
      {
        "extensions": [".js", ".jsx"]
      }
    ],
    "prettier/prettier": [
      "error",
      {
        "trailingComma": "es5",
        "singleQuote": true,
        "printWidth": 100
      }
    ]
  },
  "plugins": ["prettier"]
};

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
for (let i = 0; i < 10; i -= i) {
  console.log(i);
}
```

```bash
eslint src
```

**What did you expect to happen?**
The code should have raised lint error

**What actually happened? Please include the actual, raw output from ESLint.**
No errors were reported

**Are you willing to submit a pull request to fix this bug?**
Not a pro :(

**Comment by mdjermanovic** [comment]
Hi @da3monHunter, thanks for the report!

`for-direction` rule does work with `+=` and `-=`, but only in some cases.

For example, the rule reports an error on the following code:

```js
/* eslint for-direction: error */

for (let i = 0; i < 10; i -= 1) {
  console.log(i);
}
```

[Online Demo Link](https://eslint.org/demo#eyJ0ZXh0IjoiLyogZXNsaW50IGZvci1kaXJlY3Rpb246IGVycm9yICovXG5cbmZvciAobGV0IGkgPSAwOyBpIDwgMTA7IGkgLT0gMSkge1xuICBjb25zb2xlLmxvZyhpKTtcbn0iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOjcsInNvdXJjZVR5cGUiOiJzY3JpcHQiLCJlY21hRmVhdHVyZXMiOnt9fSwicnVsZXMiOnsiZm9yLWRpcmVjdGlvbiI6Mn0sImVudiI6e319fQ==)

This rule does not report any errors when an identifier is on the right side of `+=` or `-=`, because it doesn't know whether the variable has a positive or negative value at the moment (that's out of scope for static code analysis). Also, it doesn't perform further analysis of the expressions, which could find out that the update clause does not move counter in any direction like in your example where `i -= i`  has the same effect as `i = 0`.

**Comment by da3monHunter** [comment]
@mdjermanovic 'out of scope for static code analysis' makes sense, as it will never know what happens with the variable on runtime. Thanks for explaining :)
